### PR TITLE
docs(datasource/aws-rds): improve readme

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Improve the `aws-rds` datasource readme:
  - Warn that the default datasource config does not work with Amazon Aurora
  - Link to docs for Amazon Aurora
  - Convert **bold headings** to actual headings
  - Put descriptions of codeblock in the codeblocks
  - Show user's configuration for "terraform + rds datasource + aurora-mysql"

## Context

A user showed their config for "terraform + rds datasource + aurora-mysql", that should probably go in the docs.

They also said that the docs don't warn that by default the aws-rds datasource does not work with Amazon Aurora. So I'm warning about that in the docs too now.

Drive-by PR, not closing any issue with this, but it does resolve a bit of the discussion:

- #19928

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/aws-rds/readme.md
Comment (HonkingGoose) [inline lib/modules/datasource/aws-rds/readme.md:103]: This is basically copy/pasted from:

- https://github.com/renovatebot/renovate/discussions/19928#discussioncomment-9868352

But I changed these bits:

- Ran the config through Prettier, and made it valid JSON
- Change the users's "old" `regexManagers` to the new term: `customManagers`

I'll need a review from someone who understand all this, to make sure it's correct. 😉

File: lib/modules/datasource/aws-rds/readme.md
Comment (HonkingGoose) [inline lib/modules/datasource/aws-rds/readme.md:105]: Need a maintainer suggestion for this line too. 😉

File: lib/modules/datasource/aws-rds/readme.md
Comment (kayman-mk) [inline lib/modules/datasource/aws-rds/readme.md:97]: issue: Just saw this by accident. `amiFilter` seems to be wrong here and a copy/paste error from the very beginning as we are updating RDS versions and not an AMI id.

Comment (HonkingGoose) [reply lib/modules/datasource/aws-rds/readme.md:97]: Can you suggest the correct code? I don't understand this enough to fix it.

And thank you for the review comment! :heart:

Comment (kayman-mk) [reply lib/modules/datasource/aws-rds/readme.md:97]: As `amiFilter` is not used in the code, I suggest to replace `amiFilter` by `rdsFilter` in the readme file.

File: lib/modules/datasource/aws-rds/readme.md
Comment (rarkins) [reply lib/modules/datasource/aws-rds/readme.md:104]: Where did this code come from?

Comment (HonkingGoose) [reply lib/modules/datasource/aws-rds/readme.md:104]: I copy/pasted it from:

- https://github.com/renovatebot/renovate/discussions/19928#discussioncomment-9868352

File: lib/modules/datasource/aws-rds/readme.md
Comment (rarkins) [inline lib/modules/datasource/aws-rds/readme.md:104]:
```suggestion
The above configuration is an example of updating an AWS RDS version inside a Terraform file, using a custom manager.
```

**Review by rarkins — APPROVED** [review]

**Pull request merged**
